source/common/router/retry_state_impl.cc

```diff
@@ -58,8 +74,20 @@ RetryStateImpl::RetryStateImpl(const RetryPolicy& route_policy,
       retriable_status_codes_(route_policy.retriableStatusCodes()),
       retriable_headers_(route_policy.retriableHeaders()),
       reset_headers_(route_policy.resetHeaders()),
-      reset_max_interval_(route_policy.resetMaxInterval()) {
-
+      reset_max_interval_(route_policy.resetMaxInterval()),
+      conn_pool_new_stream_with_early_data_and_alt_svc_(
+          conn_pool_new_stream_with_early_data_and_alt_svc) {
+  if (conn_pool_new_stream_with_early_data_and_alt_svc_ &&
+      (cluster.features() & Upstream::ClusterInfo::Features::HTTP3) &&
+      Http::Utility::isSafeRequest(request_headers)) {
+    // Because 0-RTT requests could be rejected because they are sent too early, and such requests
+    // should always be retried, setup retry policy for 425 response code for all potential 0-RTT
+    // requests even though the retry policy isn't configured to do so. Since 0-RTT safe requests
+    // traditionally shouldn't have body, automatically retrying them will not cause extra
```

Comment:
They can have body though, right? We don't block that. Will that cause any issues?